[VSC-39] Implement prepare rename and rename

heitor-lassarote opened the pull request:
This PR implements the _prepare rename_ and _rename_ requests and adds tests to them. This PR also fixes a bug with finding the definitions of `ID` nodes when they are types.

heitor-lassarote committed:
[VSC-39] Implement prepare rename

Problem: Before we implement rename, we need prepare rename. This
request checks the validity of a rename.

Solution: Implement a simple handler that checks whether the position is
an ID node, that is not a keyword, or type name.

heitor-lassarote committed:
[VSC-39] Implement rename

Problem: Now that prepare rename is implemented, we can implement
rename.

Solution: This request is simple: we find all references of whatever is
under the cursor, and create an edit for all of them. We add tests to
ensure this capability works as intended, by reusing both the
definitions and prepare rename tests.

heitor-lassarote committed:
[VSC-39] Fix failing test

Problem: When searching for an ID, we always assume it's a variable.
However, it could be a type.

Solution: For now, check if we are in a `TypD` and return a `'type'` if
so. This is not perfect, but gets the test to pass.

alexd1971 commented on src/server/handlers/onReferences.ts:
```diff
@@ -0,0 +1,151 @@
+import { getContext } from '../context';
```
Perhaps extraction `onReferences` handler into separate file would be better to do in separate commit since it doesn't look relevant  here.

alexd1971 reviewed (approved):
Everything looks good to me. Just several questions

heitor-lassarote committed:
[VSC-39] Separate references handler

Problem: We need the references handler to implement rename. As it is
now, we can't easily call it, so let's refactor it out, into its own
file.

Solution: Copy and paste the references handler into a new file.
